### ISSUE TITLE
JavaScript exception in production builds (`TypeError: Ze is not a constructor`)

### ISSUE THREAD

**Issue by strarsis** [fanout/reconnecting-eventsource#56]
@harmony7:
While the development version runs without issues, a JavaScript exception occurs in the production build (TypeScript to JavaScript; `vite`):
```ts
import ReconnectingEventSource from "reconnecting-eventsource";

const eventSource = new ReconnectingEventSource(sseUrl);
````
````
TypeError: Ze is not a constructor
    at Kt (index.48de0eb5.js:1:6829)
    at index.48de0eb5.js:1:7281
    at GW (vendor.d9265921.js:52:6670)
    at e.unstable_runWithPriority (vendor.d9265921.js:37:3784)
    at bs (vendor.d9265921.js:48:38611)
    at pi (vendor.d9265921.js:52:6134)
    at r1 (vendor.d9265921.js:50:10803)
    at vendor.d9265921.js:48:38833
    at e.unstable_runWithPriority (vendor.d9265921.js:37:3784)
    at bs (vendor.d9265921.js:48:38611)
````

**Comment by strarsis** [comment]
I have to use this explicit import to get around the error in production builds:
```ts
import ReconnectingEventSource from 'reconnecting-eventsource/src/';
````
Importing from `reconnecting-eventsource/src/`, not from `reconnecting-eventsource`.

**Comment by jeremyben** [comment]
Same here with cryptic "$4 is not a constructor".
It was painful to locate the problem in minified production code !

**Comment by harmony7** [comment]
I was finally able to figure this out.

The problem happens because the name `EventSource` was being referenced using a direct global name. The code bundler that built your production code likely saw this reference and thought it was an implicitly declared variable, and gave it a short name.

I've made a change to the code and released it as 1.4.0 so that this value is referenced as a property of the global object. Now this name should not be modified by any code bundlers.

Please give it a try and let me know if this fixes it for you.

I'm going to close this for now but feel free to reopen if you still have problems.  In that case I may need to ask you for a bit more about your environment so I can set up a more similar repro, such as what bundler(s)/minifier(s) you are using.
Thanks!

**Comment by mandray** [comment]
This did not work for me in Vite using reconnecting-eventsource v1.4.0
It still shows that same is not a constructor error

**Comment by strarsis** [comment]
@mandray: Does [this workaround](https://github.com/fanout/reconnecting-eventsource/issues/56#issuecomment-1071633268) fixes the problem?

**Comment by mandray** [comment]
> @mandray: Does [this workaround](https://github.com/fanout/reconnecting-eventsource/issues/56#issuecomment-1071633268:~:text=import%20ReconnectingEventSource%20from%20%27reconnecting%2Deventsource/src/%27%3B) fix the problem?

Yes this seems to be temporarily fixing it :)
Thanks

**Comment by harmony7** [comment]
I've released a 1.5.0 that I believe will fix this.  Can you try it?
